pkg/detectors/azurecontainerregistry/azurecontainerregistry.go

```diff
@@ -35,59 +41,75 @@ func (s Scanner) Keywords() []string {
 	return []string{".azurecr.io"}
 }
 
+func (s Scanner) Type() detectorspb.DetectorType {
+	return detectorspb.DetectorType_AzureContainerRegistry
+}
+
+func (s Scanner) Description() string {
+	return "Azure's container registry is used to store docker containers. An API key can be used to override existing containers, read sensitive data, and do other operations inside the container registry."
+}
+
 // FromData will find and optionally verify Azurecontainerregistry secrets in a given set of bytes.
 func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (results []detectors.Result, err error) {
+	logger := logContext.AddLogger(ctx).Logger().WithName("azurecr")
 	dataStr := string(data)
 
-	urlMatches := url.FindAllStringSubmatch(dataStr, -1)
-	passwordMatches := password.FindAllStringSubmatch(dataStr, -1)
-
-	for _, urlMatch := range urlMatches {
-		for _, passwordMatch := range passwordMatches {
-
-			endpoint := urlMatch[0]
-			username := urlMatch[1]
-			password := passwordMatch[0]
+	// Deduplicate matches.
+	registryMatches := make(map[string]struct{})
+	for _, matches := range urlPat.FindAllStringSubmatch(dataStr, -1) {
+		u := matches[1]
+		// Ignore https://learn.microsoft.com/en-us/azure/container-registry/container-registry-private-link
+		if u == "privatelink" || u == "myacr" {
+			continue
+		}
+		registryMatches[u] = struct{}{}
+	}
+	passwordMatches := make(map[string]struct{})
+	for _, matches := range passwordPat.FindAllStringSubmatch(dataStr, -1) {
+		p := matches[0]
+		if detectors.StringShannonEntropy(p) < 4 {
+			continue
+		}
+		passwordMatches[p] = struct{}{}
+	}
 
-			s1 := detectors.Result{
+EndpointLoop:
+	for username := range registryMatches {
+		for password := range passwordMatches {
+			r := detectors.Result{
 				DetectorType: detectorspb.DetectorType_AzureContainerRegistry,
-				Raw:          []byte(endpoint),
-				RawV2:        []byte(endpoint + password),
-				Redacted:     endpoint,
+				Raw:          []byte(password),
+				RawV2:        []byte(`{"username":"` + username + `","password":"` + password + `"}`),
+				Redacted:     username,
 			}
 
 			if verify {
+				if invalidHosts.Exists(username) {
+					logger.V(3).Info("Skipping invalid registry", "username", username)
+					continue EndpointLoop
+				}
```

Comment:
I don't think there's an established 'best practice' for skipping values like this.

Should it be returned as invalid? Should it be skipped altogether? etc.

Reply:
I'd lean towards skipping it. Returning it as an invalid result might not add much value, but I don't have a strong preference